demo/fancy_chat.py

```diff
@@ -550,6 +554,13 @@ def on_click_regenerate(e: me.ClickEvent):
   yield
 
 
+def on_submit_chat_msg(e: me.TextareaShortcutEvent):
+  state = me.state(State)
+  state.input = e.value
+  yield
+  yield from on_click_submit_chat_msg(me.ClickEvent(key=e.key, is_target=False))
```

Comment:
this works, but it's a little weird of a pattern (for humans and LLM :), I think it's cleaner to have a new function "submit_chat_msg" and have both this function and the on_click function yield from that common function. This way you don't need to construct a click event.